code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -569,3 +569,96 @@
 		M.Jitter(4)
 		M.Dizzy(4)
 	..()
+
+
+/// Can bring a corpse back to life temporarily (if heart is intact)
+/// Also prevents dying
+/datum/reagent/drug/nooartrium
+	name = "Nooartrium"
+	description = "A reagent that is known to stimulate the heart in a dead patient, temporarily bringing back recently dead patients at great cost to their heart."
+	overdose_threshold = 25
+	self_consuming = TRUE //No pesky liver shenanigans
+	///If we brought someone back from the dead
+	var/back_from_the_dead = FALSE
+	var/consequences  = FALSE // This can't be healthy?
+
+
+/datum/reagent/drug/nooartrium/on_mob_add(mob/living/affected_mob)
+	if(affected_mob.suiciding)
+		return
+	var/obj/item/organ/heart/heart = affected_mob.getorganslot(ORGAN_SLOT_HEART)
+	if(!heart || heart.organ_flags & ORGAN_FAILING)
+		return
+	ADD_TRAIT(affected_mob, TRAIT_NOCRITDAMAGE, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NODEATH, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOHARDCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOSOFTCRIT, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_STABLEHEART, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOLIMBDISABLE, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_STUNRESISTANCE, FROM_NOOARTRIUM)
+	ADD_TRAIT(affected_mob, TRAIT_NOSTAMCRIT, FROM_NOOARTRIUM) // Moving corpses don't get tired
+	ADD_TRAIT(affected_mob, TRAIT_IGNOREDAMAGESLOWDOWN, FROM_NOOARTRIUM)
+	if(affected_mob.stat == DEAD)
+		back_from_the_dead = TRUE
+	affected_mob.set_stat(CONSCIOUS) // This doesn't touch knocked out
+	affected_mob.updatehealth()
+	affected_mob.update_sight()
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, STAT_TRAIT)
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, CRIT_HEALTH_TRAIT) // Normally updated using set_health() - we don't want to adjust health, and NOHARDCRIT blocks it being re-added, but not removed
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT) // Prevents knockout by oxyloss
+	affected_mob.set_resting(FALSE) // Please get up. No one wants a death throes juggernaut lying on the floor
+	affected_mob.SetAllImmobility(0)
+	if(!back_from_the_dead)
+		to_chat(affected_mob, span_userdanger("You feel your heart start beating with incredible strength!"))
+		return
+	affected_mob.grab_ghost(force = FALSE) //Shoves them back into their freshly reanimated corpse.
+	affected_mob.emote("gasp")
+	to_chat(affected_mob, span_userdanger("You feel your heart start beating with incredible strength, forcing your battered body to move!"))
+
+
+/datum/reagent/drug/nooartrium/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
+	. = ..()
+	if (!consequences && times_fired > 210)
+		consequences = TRUE
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, CRIT_HEALTH_TRAIT)
+	REMOVE_TRAIT(affected_mob, TRAIT_KNOCKEDOUT, OXYLOSS_TRAIT)
+	affected_mob.adjustBruteLoss(0.25 * delta_time)
```

Comment:
```suggestion
```

This is not necessary with limb damage re-enabled, in fact it will cause the damage to be spread evenly over the whole body potentially inflicting limb disability prematurely.